src/ImageSharp/Formats/Tiff/PhotometricInterpretation/YCbCrConverter.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Runtime.CompilerServices;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.Formats.Tiff.PhotometricInterpretation
+{
+    /// <summary>
+    /// Converts YCbCr data to rgb data.
+    /// </summary>
+    internal class YCbCrConverter
+    {
+        private readonly CodingRangeExpander yExpander;
+        private readonly CodingRangeExpander cbExpander;
+        private readonly CodingRangeExpander crExpander;
+        private readonly YCbCrToRgbConverter converter;
+
+        private static readonly Rational[] DefaultLuma =
+        {
+            new Rational(299, 1000),
+            new Rational(587, 1000),
+            new Rational(114, 1000)
+        };
+
+        private static readonly Rational[] DefaultReferenceBlackWhite =
+        {
+            new Rational(0, 1), new Rational(255, 1),
+            new Rational(128, 1), new Rational(255, 1),
+            new Rational(128, 1), new Rational(255, 1)
+        };
+
+        public YCbCrConverter(Rational[] referenceBlackAndWhite, Rational[] coefficients)
+        {
+            referenceBlackAndWhite ??= DefaultReferenceBlackWhite;
+            coefficients ??= DefaultLuma;
+
+            if (referenceBlackAndWhite.Length != 6)
+            {
+                TiffThrowHelper.ThrowImageFormatException("reference black and white array should have 6 entry's");
+            }
+
+            if (coefficients.Length != 3)
+            {
+                TiffThrowHelper.ThrowImageFormatException("luma coefficients array should have 6 entry's");
+            }
+
+            this.yExpander = new CodingRangeExpander(referenceBlackAndWhite[0], referenceBlackAndWhite[1], 255);
+            this.cbExpander = new CodingRangeExpander(referenceBlackAndWhite[2], referenceBlackAndWhite[3], 127);
+            this.crExpander = new CodingRangeExpander(referenceBlackAndWhite[4], referenceBlackAndWhite[5], 127);
+            this.converter = new YCbCrToRgbConverter(coefficients[0], coefficients[1], coefficients[2]);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Rgba32 ConvertToRgba32(byte y, byte cb, byte cr)
+        {
+            float yExpanded = this.yExpander.Expand(y);
+            float cbExpanded = this.cbExpander.Expand(cb);
+            float crExpanded = this.crExpander.Expand(cr);
+
+            Rgba32 rgba = this.converter.Convert(yExpanded, cbExpanded, crExpanded);
+
+            return rgba;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static byte RoundAndClampTo8Bit(float value)
+        {
+            int input = (int)MathF.Round(value);
+            return (byte)Math.Min(Math.Max(input, 0), 255);
```

Comment:
```suggestion
            return (byte)Math.Clamp(input, 0, 255);
```
? (and if it's available for all TFMs).

Right now codegen is pretty much equal, but in a future JIT there could be some improvements in that area.

Reply:
yes makes sense, thanks

Reply:
unfortunately not available with `net472`, using `Numerics.Clamp` now.

Reply:
Numerics version is faster than the runtime one I think

Reply:
> Numerics version is faster than the runtime one I think

The runtime one has an additional check for min > max which isn't needed here as the limits are constants (0 and 255). After inlineing the JIT is able to remove that min > max check, so codegen is quite similar. The difference is that the Numerics-version checks value > max first, while the runtime-version checks value < min first. So which version is faster depends on the likeness of value < min vs value > max and the branch-predictor's history (and potentially on dynamic profile guided optimization).

`Numerics.Clamp` is good to use here. Maybe just add a `Debug.Assert(min < max)` there to catch "typos" early?